test/odbc/test_pg_datatype.cpp

```diff
@@ -0,0 +1,160 @@
+  odbcHandler.CloseStmt();
+
+  // Step 3: Select all from the tables and assert that the following attributes of the type is correct:
+  odbcHandler.ExecQuery(SelectStatement(TABLE_NAME, {"*"}, vector<string> {"pk"}));
+
+  // Make sure attributes are correct
+  rcode = SQLColAttribute(odbcHandler.GetStatementHandle(),
+                          2,
+                          SQL_DESC_LENGTH,
+                          NULL,
+                          0,
+                          NULL,
+                          (SQLLEN*) &length); // not the same datatype as documented?
+  ASSERT_EQ(rcode, SQL_SUCCESS);
+  ASSERT_EQ(length, LENGTH_EXPECTED);
+  
+  rcode = SQLColAttribute(odbcHandler.GetStatementHandle(),
+                          2,
+                          SQL_DESC_PRECISION,
+                          NULL,
+                          0,
+                          NULL,
+                          (SQLLEN*) &precision); 
+  ASSERT_EQ(rcode, SQL_SUCCESS);
+  ASSERT_EQ(precision, PRECISION_EXPECTED);
+
+  rcode = SQLColAttribute(odbcHandler.GetStatementHandle(),
+                          2,
+                          SQL_DESC_SCALE,
+                          NULL,
+                          0,
+                          NULL,
+                          (SQLLEN*) &scale); 
+  ASSERT_EQ(rcode, SQL_SUCCESS);
+  ASSERT_EQ(scale, SCALE_EXPECTED);
+
+  rcode = SQLColAttribute(odbcHandler.GetStatementHandle(),
+                          2,
+                          SQL_DESC_TYPE_NAME,
+                          name,
+                          BUFFER_SIZE,
+                          (SQLSMALLINT*) &strlength,
+                          NULL); 
+  ASSERT_EQ(string(name), NAME_EXPECTED);
+
+  // Make sure values are correct
+  ASSERT_NO_FATAL_FAILURE(odbcHandler.BindColumns(bind_columns));
+
+  for (int i = 0; i < NUM_ROWS; ++i) {
+    
+    rcode = SQLFetch(odbcHandler.GetStatementHandle());
+    ASSERT_EQ(rcode, SQL_SUCCESS);
+    ASSERT_EQ(pk_len, BYTES_EXPECTED);
+    ASSERT_EQ(pk, i);
+
+    if (inserted_values[i] != "NULL")
+    {
+      ASSERT_EQ(data_len, BYTES_EXPECTED);
+      ASSERT_EQ(data, strtoll(inserted_values[i].c_str(), NULL, 10));
+    } 
+    else 
+      ASSERT_EQ(data_len, SQL_NULL_DATA);
+  }
+
+  rcode = SQLFetch(odbcHandler.GetStatementHandle());
+  ASSERT_EQ(rcode, SQL_NO_DATA);
+  odbcHandler.CloseStmt();
+
+  // Step 4: Update values of one of the rows and assert that the value has been updated by selecting it.
+    // TODO: Make an update statement function in query_generator.cpp/.h?
+  odbcHandler.ExecQuery("UPDATE " + TABLE_NAME + " SET " + COL2_NAME + " = 100 WHERE " + COL1_NAME + " = 3"); 
+
+  odbcHandler.CloseStmt();
+  odbcHandler.ExecQuery(SelectStatement(TABLE_NAME, {"*"}, vector<string>{}, "pk = 3 "));
+
+  ASSERT_NO_FATAL_FAILURE(odbcHandler.BindColumns(bind_columns));
+
+  rcode = SQLFetch(odbcHandler.GetStatementHandle());
+
+  ASSERT_EQ(rcode, SQL_SUCCESS);
```

Comment:
Is it possible to check if only 1 row was affected by this UPDATE?

Reply:
Good question, I think I can use SQLRowCount() so I'll do that.